tsup-lib.config.ts

```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from "tsup"
+
+export default defineConfig({
+  entry: ["./lib/index.ts"],
+  format: ["esm"],
+  sourcemap: "inline",
+  outDir: "dist/lib",
+  noExternal: ["@tscircuit/parts-engine"],
```

Comment:
why isn't this needed for other modules?

Reply:
Maybe the build config of parts-engine has some issue